src/QuickEspNow_esp32.cpp

```diff
@@ -259,6 +259,10 @@ bool QuickEspNow::addPeer (const uint8_t* peer_addr) {
     auto peer = esp_now_peer_info_t{};
     esp_err_t error = ESP_OK;
 
+    if (peer_list.peer_exists (peer_addr) && !esp_now_is_peer_exist(peer_addr)) {
```

Comment:
Is there more sanity checking that needs to happen here? In line 268 they call ESP_ERROR_CHECK. What does that check for?

Reply:
That function is used when user wants to sort of "throw" an exception when the returned esp_err_t is not ESP_OK. Line 268 is actually causing the core to panic and restart. So its used as a last resort to prevent memory failures. The function I added safely checks if the esp now has such peer stored, thus preventing that crash

Reply:
But is this the only sanity checking that needs to happen, or is there more that would require deleting the peer if it exists?

Reply:
I checked how the peers are added to the buffer in this function, the checks for that are already there and there aren't any other  calls that add to the buffer. But for some reason I couldn't figure, the esp now itself thinks the peer is not added while the quick esp now buffer has the peer stored.

Reply:
> But is this the only sanity checking that needs to happen

For the sake of this problem after quickly skimming through the esp now api, I think so yes.